### PR TITLE
Add advanced error handling w/ ruby

### PR DESCRIPTION
**Pull request by bobbytables**
Adds how to encode / decode details into errors with go / ruby.

### CODE REVIEW

**Review by jhump — APPROVED** [review]
Sweet. Thanks for adding this!

File: examples/ruby/client_with_error.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+$LOAD_PATH.unshift File.expand_path("../proto", __FILE__)
+require_relative './proto/swapi_pb'
+require_relative './proto/swapi_services_pb'
+require 'byebug'
```
Comment (pseudomuto) [inline examples/ruby/client_with_error.rb]: Leftover from testing?

**Review by pseudomuto — APPROVED** [review]
LGTM 🐑 🇮🇹

File: examples/ruby/client_with_error.rb (hunk 1 above)
Comment (bobbytables) [reply examples/ruby/client_with_error.rb]: Whoops, nice catch.

**Comment by bobbytables** [comment]
I gotta say, this was incredibly un-documented by gRPC. The ruby usage is _buried_ in tests. Super hard to find.

**Review by carlessistare — APPROVED** [review]

**Pull request merged**